server/util/perms/perms.go

```diff
@@ -28,6 +28,24 @@ const (
 	OTHERS_WRITE = 0o02
 	OTHERS_EXEC  = 0o01
 	ALL          = 0o0777
+
+	// Constants for UserGroup.Role. These are powers of 2 so that we can allow
+	// assigning multiple roles to users and use these as bitmasks to check
+	// role membership.
+
+	// DeveloperRole means a user cannot perform certain privileged actions such
+	// as creating API keys and viewing usage data, but can perform most other
+	// common actions such as viewing invocation history.
+	DeveloperRole = 0
+	// AdminRole means a user has unrestricted access within a group.
+	AdminRole = 1
+
+	// DefaultRole is the role assigned to users when joining a group they did
+	// not create.
+	// TODO(bduffany): Change this to DeveloperRole once we have a way to manage
+	// roles via the UI (otherwise, there would be no easy way to promote new
+	// users to admins in the meantime).
+	DefaultRole = AdminRole
 )
```

Comment:
nit: can you move these to their own const () block, just for clarity?

You might also consider, for example:

type role uint32

const (
  DeveloperRole role = 1 << 0
  AdminRole role = 1 << 1
  ManagerRole role = 1 << 2
  BadRole role = 1 << 100 // <- this line won't compile, because it overflows 'role'
)

Reply:
Done